docs/doc/60-contributing/03-rfcs/20221013-iceberg-external-table.md

```diff
@@ -0,0 +1,203 @@
+---
+
+- RFC PR: [datafuselabs/databend#8215](https://github.com/datafuselabs/databend/pull/8215)
+- Tracking Issue: [datafuselabs/databend#8216](https://github.com/datafuselabs/databend/issues/8216)
+
+## Summary
+
+Iceberg is a widely supported table format among data lake-houses.
+This RFC describes how the iceberg external table will behave and how will we implement it.
+
+## Motivation
+
+The [Apache Iceberg](https://iceberg.apache.org) is a table format widely used in data lake-houses, it offers a more complete logical view of databases and has much higher performance of accessing as external tables, since users don't need to know about partitions and less indexing through files when accessing is required comparing to `Hive`.
+
+The splitted and well-defined structure of `Iceberg` also makes concurrent accesssing and version managementing on data sources safer, saver and convienient.
+
+![stack of Apache Iceberg](https://iceberg.apache.org/img/iceberg-metadata.png)
+
+Supporting Iceberg will empower databend as a open data lake, giving it more OLAP abilities.
+
+## Guide-level explanation
+
+### Create Iceberg Table
+
+To use Iceberg external table, users need to have an iceberg storage with neccessary permission established. Then they can establish a external table with `ICEBERG` engine:
+
+```sql
+CREATE EXTERNAL TABLE [IF NOT EXISTS] [db.]table_name
+(
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
+    ...
+) ENGINE=ICEBERG
+ENGINE_OPTION=(
+  DATABASE='db0'
+  TABLE='tbl0'
+)
+<external-location>
+```
+
+where:
+
+- data_type:
+
+```
+<data_type>:
+  TINYINT
+| SMALLINT
+| INT
+| BIGINT
+| FLOAT
+| DOUBLE
+| DATE
+| TIMESTAMP
+| VARCHAR
+| ARRAY
+| OBJECT
+| VARIANT
+```
+
+- external-location (using S3 for example):
+
+```
+external-location ::=
+
+URL = 's3://<bucket-name>/<path-to-iceberg>'
+CONNECTION = (
+  ENDPOINT_URL = <endpoint_url>
+  ACCESS_KEY_ID = <access_key_id>
+  ACCESS_KEY_SECRET = <access_key_secret>
+  SESSION_TOKEN = <aws_session_token>
+)
+```
+
+### Accessing Iceberg Table's content
+
+```sql
+-- assuming users have created an external table named `iceberg_tbl`
+SELECT COUNT(*) FROM iceberg_tbl;
+```
+
+Joint query on normal table and Iceberg Table:
+
+```sql
+SELECT normal_tbl.book_name, iceberg_tbl.author FROM normal_tbl, iceberg_tbl WHERE normal_tbl.isbn = iceberg_tbl.isbn AND iceberg_tbl.sales > 100000;
+```
+
+On operating the table, all data remains still on the user-provided ends.
+
+### Time Travel
+
+Iceberg offers a list of snapshots and its timestamps. Time travelling on it is natrually.
+
+```sql
+SELECT ...
+FROM iceberg_table
+AT ( { SNAPSHOT => <snapshot_id> | TIMESTAMP => <timestamp> } );
+```
+
+#### Accessing Snapshot Metadata
+
+Users should be able to lookup the list of snapshot id in Iceberg:
+
+```sql
+SELECT snapshot_id FROM FUSE_SNAPSHOT('<database-name>', '<external-iceberg-table-name>');
```

Comment:
`FUSE_SNAPSHOT` should be only provided by `fuse engine`? @dantengsky

Reply:
i guess add a seperate ICEBERG_SNAPSHOT might better